make/modules/java.desktop/lib/Awt2dLibraries.gmk

```diff
@@ -570,6 +570,7 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBFONTMANAGER, \
     CFLAGS_windows = -DCC_NOEX, \
     EXTRA_HEADER_DIRS := $(LIBFONTMANAGER_EXTRA_HEADER_DIRS), \
     WARNINGS_AS_ERRORS_xlc := false, \
+    DISABLED_WARNINGS_clang := deprecated-declarations, \
```

Comment:
I see this added here and in another location. What is deprecated ? You really need to explain these sorts of things.
I've built JDK using Xcode 12.3 and not had any need for this.

Reply:
Hello
I believe it was a workaround for issues with xcode 12.2 in early beta days.
Those issues were fixed later in upstream jdk, so most likely we need to remove these workarounds.

Reply:
It seems these workarounds are still needed:

```
jdk/src/java.desktop/macosx/native/libsplashscreen/splashscreen_sys.m:300:39: error: 'NSAlphaFirstBitmapFormat' is deprecated: first deprecated in macOS 10.14 [-Werror,-Wdeprecated-declarations]
                        bitmapFormat: NSAlphaFirstBitmapFormat | NSAlphaNonpremultipliedBitmapFormat
                                      ^~~~~~~~~~~~~~~~~~~~~~~~
                                      NSBitmapFormatAlphaFirst

jdk/src/java.desktop/macosx/native/libsplashscreen/splashscreen_sys.m:438:34: error: 'NSBorderlessWindowMask' is deprecated: first deprecated in macOS 10.12 [-Werror,-Wdeprecated-declarations]
                      styleMask: NSBorderlessWindowMask
                                 ^~~~~~~~~~~~~~~~~~~~~~
                                 NSWindowStyleMaskBorderless

```

Reply:
Are you doing something somewhere to change the target version of macOS or SDK ? I had no such problem.
I think we currently target a macOS 10.9 and if you are changing that it would need discussion.
If you are changing it only for Mac ARM that may make more sense .. 

And these appear to be just API churn by Apple
NSAlphaFirstBitmapFormat is replaced by NSBitmapFormatAlphaFirst

https://developer.apple.com/documentation/appkit/nsbitmapformat/nsbitmapformatalphafirst?language=objc

NSBorderlessWindowMask is replaced by NSWindowStyleMask

https://developer.apple.com/documentation/appkit/nswindowstylemask?language=occ

Reply:
Min_macos version is changed to 11.0 for macos_aarch64

https://github.com/openjdk/jdk/pull/2200/files/0c2cb0a372bf1a8607810d773b53d6959616a816#diff-7cd97cdbeb3053597e5d6659016cdf0f60b2c412bd39934a43817ee0b717b7a7R136

Reply:
1) I meant change to NSWindowStyleMaskBorderless from NSBorderlessWindowMask
2) So maybe rather than the deprecation suppression  you could change both constants to the new ones.
Ordinarily I'd say let someone else do that but this looks like a simple obvious change and is dwarfed by all the other changes going on for Mac ARM ...

Reply:
that sounds good to me, I am just afraid to break intel mac on older macos versions with this change.

Reply:
That may actually be a valid concern. Both say macOS 10.12+ ...  which might conflict with the 10.9 target.

Reply:
Maybe you should just file a bug after all for this to be dealt with separately.
Certainly if it is NOT fixed now such a bug needs to be filed.

Reply:
I have created https://bugs.openjdk.java.net/browse/JDK-8260402 which is blocked by jep-391 currently